tiledb/cc/attribute.cc

```diff
@@ -19,28 +19,42 @@ void set_fill_value(Attribute &attr, py::array value) {
 }
 
 py::array get_fill_value(Attribute &attr) {
+  // Get the fill value from the C++ API as a void* value.
   const void *value;
   uint64_t size;
-
   attr.get_fill_value(&value, &size);
 
-  auto value_num = attr.cell_val_num();
-  auto value_type = tdb_to_np_dtype(attr.type(), value_num);
-
+  // If this is a string type, we want to return each value as a single cell.
   if (is_tdb_str(attr.type())) {
-    value_type = py::dtype("|S1");
-    value_num = size;
+    auto value_type = py::dtype("|S1");
+    return py::array(value_type, size, value);
   }
 
-  // record type
+  // If this is a record type (void), return a single cell.
+  // If this is a blob-like type, we want to return each value as a single byte
+  // cell.
+  auto tdb_type = attr.type();
+  if (tdb_type == TILEDB_BLOB
+#if TILEDB_VERSION_MAJOR >= 2 && TILEDB_VERSION_MINOR >= 21
```

Comment:
if there's a tiledb 3 some day:
```suggestion
#if TILEDB_VERSION_MAJOR > 2 || (TILEDB_VERSION_MAJOR == 2 && TILEDB_VERSION_MINOR >= 21)
```